File: app/src/main/java/com/italankin/dictionary/dto/Language.java
package com.italankin.dictionary.dto;

import androidx.annotation.NonNull;

/**
 * Class for handling languages data.
 */
public class Language implements Comparable<Language> {

    private String code;
    private String name;

    public Language(String code, String name) {
        this.code = code;
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public String getCode() {
        return code;
    }

    @Override
    public int compareTo(@NonNull Language another) {
        return this.name.compareTo(another.name);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        return code.equals(((Language) o).code);
    }

    @Override
    public int hashCode() {
        return code.hashCode();
    }
}

File: app/src/main/java/com/italankin/dictionary/api/dto/Language.kt
package com.italankin.dictionary.api.dto

/**
 * Class for handling languages data.
 */
class Language(
        val code: String,
        val name: String
) : Comparable<Language> {

    override fun compareTo(other: Language) = name.compareTo(other.name)

    override fun hashCode() = code.hashCode()

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Language) return false
        return code == other.code
    }
}
